lib/Screens/Reading/reading_screen.dart

```diff
@@ -18,43 +17,57 @@ class ReadingScreen extends StatelessWidget {
   final String title;
   final String image;
   final String id;
-  final DatabaseMethods databaseMethods = new DatabaseMethods();
+
+  @override
+  State<ReadingScreen> createState() => _ReadingScreenState();
+}
+
+class _ReadingScreenState extends State<ReadingScreen> {
+  late String userEmail;
+  String ratingTitle = 'Avalie este conteúdo!';
+  double initialRating = 0.0;
+  String commentHint = 'Nos conte o que achou!';
+
+  @override
+  void initState() {
+    getUserEmail();
+    super.initState();
+  }
+
+  getUserEmail() async {
+    await HelperFunctions.getUserEmailInSharedPreference().then((email) {
+      setState(() {
+        userEmail = email;
+      });
+    });
+  }
 
   // Cria um dialogo para a avaliacao de uma intervencaos
   void _showRatingDialog(context, String dialogTitle, String id) async {
-    String formattedDate = DateFormat.yMEd().add_jms().format(DateTime.now());
-    bool existingRating = false;
-    var ds = await DatabaseMethods().ratingsAreEmpty(id);
-    String ratingTitle = 'Avalie este conteúdo!';
-
-    existingRating = ds.docs.length != 0;
-    if (existingRating) {
-      ratingTitle =
-          'Você já avaliou este conteúdo, deseja avaliá-lo novamente?';
-    }
+    await UserService().findReadingRating(userEmail, id).then((readingRating) {
+      if (readingRating.id != null) {
```

Comment:
Pq o id que tem que ser diferente de nulo, há a possibilidade do readingRating nao vier nulo e mesmo assim não satisfazer essa condicional?

Reply:
Se for nulo quer dizer que não encontrou nada no banco de dados, se for diferente de nulo existe uma avaliação da leitura no banco de dados e ele usa os valores encontrados

Reply:
mas pode ser só o readingRating != null, nao?

Reply:
não, dá um erro falando: The operand can't be null, so the condition is always true.